OWML.ModLoader/ModData.cs

```diff
@@ -11,134 +11,34 @@ public class ModData : IModData
     {
         public IModManifest Manifest { get; }
         public IModConfig Config { get; private set; }
+        [Obsolete("Maybe this should be obsolete?")]
         public IModConfig DefaultConfig { get; private set; }
+        public IModConfig UserConfig { get; private set; }
         public bool RequireReload => Config.Enabled != _configSnapshot.Enabled;
         public bool RequireVR => Manifest.RequireVR;
 
-        public bool Enabled => Config != null && Config.Enabled ||
-                               Config == null && DefaultConfig.Enabled;
+        public bool Enabled => Config.Enabled;
+
 
         private IModConfig _configSnapshot;
 
         public ModData(IModManifest manifest, IModConfig config, IModConfig defaultConfig)
         {
             Manifest = manifest;
-            Config = config;
-            DefaultConfig = defaultConfig ?? new ModConfig();
+            Config = new ModMergedConfig(config, defaultConfig, manifest);
```

Comment:
Config -> MergedConfig?

Reply:
I was trying to keep it as Config to preserve backwards compatibility. Does that not matter here? I guess mods aren't supposed to access this directly.

This code could be a lot cleaner if ModMergedConfig didn't have to be of type IModConfig, so if I don't have to worry about backwards compatibility with Config I can clean this up a lot.

Reply:
I mean ModMergedConfig would still implement IModConfig, but in ModData it would be declared as type ModMergedConfig, so that I can call ModMergedConfig specific methods, no need to do some of the workarounds I'm doing now. But only if backwards compatibility doesn't matter here.

Reply:
from memory, whats importantnisnthat mod.Configure is called with IModConfig. that might be the only thing that matters. 
General tip: experiment, test with all mods. No errors -> it's fine ;-)

Reply:
Made the change and tested with all mods. Had to fix an error with missing configs, other than that seems to work fine :)

Reply:
I left it as "Config" since now there's no UserConfig. Should I change it to MergedConfig anyway?